modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/messaging/test/IndividualSegmentsCheckerTest.java

```diff
@@ -0,0 +1,200 @@
+
+	@Before
+	public void setUp() throws Exception {
+		_setIndividualSegmentsCheckerTest();
+	}
+
+	@Test(expected = Test.None.class)
+	public void testCheckIndividualSegments() throws Exception {
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId",
+							AnalyticsConnectionPropertiesConstants.
+								LIFERAY_ANALYTICS_DATA_SOURCE_ID
+						).put(
+							"liferayAnalyticsEnableAllGroupIds",
+							AnalyticsConnectionPropertiesConstants.
+								LIFERAY_ANALYTICS_ENABLE_ALL_GROUP_IDS
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							AnalyticsConnectionPropertiesConstants.
+								LIFERAY_ANALYTICS_FARO_BACKEND_SECURITY_SIGNATURE
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							AnalyticsConnectionPropertiesConstants.
+								LIFERAY_ANALYTICS_FARO_BACKEND_URL
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			ReflectionTestUtil.invoke(
+				_individualSegmentsChecker, "checkIndividualSegments",
+				new Class<?>[0]);
+		}
+	}
+
+	private Object _getInstance(
+			String className, Bundle segmentsAsahConnectorBundle)
+		throws Exception {
+
+		Class<?> clazz = segmentsAsahConnectorBundle.loadClass(className);
+
+		Constructor<?> constructor = clazz.getConstructor();
+
+		return constructor.newInstance();
+	}
+
+	private void _setIndividualSegmentsCheckerTest() throws Exception {
+		Bundle bundle = FrameworkUtil.getBundle(
+			IndividualSegmentsCheckerTest.class);
+
+		BundleContext bundleContext = bundle.getBundleContext();
+
+		Bundle segmentsAsahConnectorBundle = null;
+
+		for (Bundle curBundle : bundleContext.getBundles()) {
+			if (Objects.equals(
+					curBundle.getSymbolicName(),
+					"com.liferay.segments.asah.connector")) {
+
+				segmentsAsahConnectorBundle = curBundle;
+
+				break;
+			}
+		}
+
+		Assert.assertNotNull(
+			"Unable to find com.liferay.segments.asah.connector bundle",
+			segmentsAsahConnectorBundle);
+
+		_individualSegmentsChecker = _getInstance(
+			"com.liferay.segments.asah.connector.internal.messaging." +
+				"IndividualSegmentsChecker",
+			segmentsAsahConnectorBundle);
+
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsChecker, "_analyticsSettingsManager",
+			_analyticsSettingsManager);
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsChecker, "_asahSegmentsEntryCache",
+			_getInstance(
+				"com.liferay.segments.asah.connector.internal.cache." +
+					"AsahSegmentsEntryCache",
+				segmentsAsahConnectorBundle));
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsChecker, "_companyLocalService",
+			_companyLocalService);
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsChecker, "_http", _http);
```

Comment:
You should create your own HTTP component, so we don't call AC, it is not possible to call AC from an Integration tests.

Take a look to com.liferay.analytics.reports.web.internal.portlet.action.test.GetHistoricalViewsMVCResourceCommandTest